Services/ManagerService.cs

```diff
@@ -0,0 +1,99 @@
+
+        public ManagerService()
+        {
+            _fileService = new FileService();
+        }
+        public List<Booking> FilterBookings(FilterCriteria criteria)
+        {
+            var bookings = LoadBookings();
+
+            return bookings.Where(b =>
+            {
+                var flight = GetFlight(b.FlightNumber);
+                var flightPrice = flight != null ? GetFlightPrice(b.FlightNumber, b.Class) : (decimal?)null;                
+                return
+                    (criteria.FlightNumber == null || b.FlightNumber == criteria.FlightNumber) &&
+                    (criteria.Price == null || flightPrice == criteria.Price) &&
+                    (criteria.DepartureCountry == null || (flight != null && flight.DepartureCountry == criteria.DepartureCountry)) &&
+                    (criteria.DestinationCountry == null || (flight != null && flight.DestinationCountry == criteria.DestinationCountry)) &&
+                    (criteria.DepartureDate == null || (flight != null && flight.DepartureDate.HasValue && flight.DepartureDate.Value.Date == criteria.DepartureDate.Value.Date)) &&
+                    (criteria.DepartureAirport == null || (flight != null && flight.DepartureAirport == criteria.DepartureAirport)) &&
+                    (criteria.ArrivalAirport == null || (flight != null && flight.ArrivalAirport == criteria.ArrivalAirport)) &&
+                    (criteria.Passenger == null || b.PassengerId == criteria.Passenger) &&
+                    (criteria.Class == null || b.Class == criteria.Class);
+            }).ToList();
+        }
+
+        public List<string> Validate()
+        {
+            var flights = LoadFlights();
+
+            FlightValidator fv = new FlightValidator();
+            List<string> result = fv.ValidateFlight(flights);
+
+            return result;
+        }
+
+        public void ValidateReport()
+        {
+            var txt = Validate();
+            foreach(var t in txt)
+            {
+                Console.WriteLine(t);
+            }
+            if(txt.Count == 0)
+            {
+                Console.WriteLine("All Flights are valid");
+            }
+        }
+
+        private void SaveFlight(Flight flight)
+        {
+            var flights = LoadFlights();
+            flights.Add(flight);
+
+            _fileService.SaveToFile(flightFilePath, flights);
+        }
+
+        private Flight GetFlight(string flightNumber)
+        {
+            var flights = LoadFlights();
+            return flights.FirstOrDefault(f => f.FlightNumber == flightNumber) ?? new Flight();
+        }
+
+        private List<Booking> LoadBookings()
+        {
+            return _fileService.LoadFromFile<Booking>(bookingFilePath);
+        }
+
+        private List<Flight> LoadFlights()
+        {
+            return _fileService.LoadFromFile<Flight>(flightFilePath);
+        }
```

Comment:
We have already separated the fileService from everything else,
Why not also separate the use of fileService and bookingFilePath too? We can use a class called DataLoader that has the paths and functions to do the loading..

Why would we do that?
* We won't need to do LoadFlights() every single time!
* We would separate the concerns, ie, a class for data retrieval, and a class for data modification
* We know for sure this service is going to be handling two files, we're not using database fetches so we're not constantly fetching data, Why not just load them once?

That way we'll have `private List<Flight> flights` and `private List<Booking> bookings` in this class